libarchive/archive_read.c

```diff
@@ -429,6 +429,7 @@ archive_read_next_header2(struct archive *_a, struct archive_entry *entry)
 	switch (ret) {
 	case ARCHIVE_EOF:
 		a->archive.state = ARCHIVE_STATE_EOF;
+		--_a->file_count;/* Revert a file counter. */
```

Comment:
Do we want the counter incremented in the `ARCHIVE_{RETRY,FATAL}` cases?  Seems to me that it might be better to perform the increment only in the `ARCHIVE_{OK,WARN}` cases.

Or do we need to increment the counter before calling `read_header` for some reason?

Reply:
I'd do it that way, but that particular commit is backported from libarchive (as was the original `file_count` commit).  I thought that it wasn't important enough to diverge from libarchive in this case.

Reply:
Ah, I didn't notice that.  Agreed, no point diverging.

Reply:
I winced a bit at the length of the commit subject line (not to mention the spelling), but again I figured it was better to avoid diverging as much as possible.  :)